python/py-requests-cache/Portfile

```diff
@@ -18,29 +17,43 @@ maintainers         nomaintainer
 description         Transparent persistent cache for py-requests
 long_description    ${description}
 
-checksums           rmd160  511d993d81abeffca4d13414a036be94c5c9ce23 \
-                    sha256  1ea8b866db2b16a9b058626c7e1bde0cf669e96abc436cdef352f60237e23ab9 \
-                    size    29527
+homepage            https://requests-cache.readthedocs.io/en/stable/
 
-python.versions     27 35 36 37 38
+checksums           rmd160  41cd04c6a75cd6d10ce08c5e3562ab4517fbe267 \
+                    sha256  27d3eb276ab3affa9864dfc0475241d6d960dd566d57ec46ffa7759c2c74ed1c \
+                    size    1510446
+
+python.versions     27 35 36 37 38 39 410
 
 if {${name} ne ${subport}} {
     depends_build-append \
                     port:py${python.version}-setuptools
 
     depends_lib-append \
-                    port:py${python.version}-requests
+                    port:py${python.version}-appdirs       \
+                    port:py${python.version}-attrs         \
+                    port:py${python.version}-cattrs        \
+                    port:py${python.version}-requests      \
+                    port:py${python.version}-url-normalize \
+                    port:py${python.version}-urllib3
 
     post-destroot {
         set docdir ${prefix}/share/doc/${subport}
         xinstall -d ${destroot}${docdir}
-        xinstall -m 0644 -W ${worksrcpath} README.rst CONTRIBUTORS.rst \
-            LICENSE HISTORY.rst ${destroot}${docdir}
+        xinstall -m 0644 -W ${worksrcpath} README.md CONTRIBUTORS.md \
+            LICENSE HISTORY.md ${destroot}${docdir}
     }
 
     depends_test-append \
-                    port:py${python.version}-mock \
-                    port:py${python.version}-pytest
+                    port:py${python.version}-redis             \
+                    port:py${python.version}-itsdangerous      \
+                    port:py${python.version}-yaml              \
+                    port:py${python.version}-ujson             \
+                    port:py${python.version}-mock              \
+                    port:py${python.version}-requests-mock     \
+                    port:py${python.version}-timeout-decorator \
+                    port:py${python.version}-pytest            \
+                    port:py${python.version}-responses
```

Comment:
after you've declared all the dependencies for the latest version you can add a block like this:
```
    if {${python.version} < 36} {
        version     0.5.2
        revision    0
        checksums   rmd160  ccd76aca28b1ba826a88beb5fc65f16aec26e583 \
                    sha256  813023269686045f8e01e2289cc1e7e9ae5ab22ddd1e2849a9093ab3ab7270eb \
                    size    31159
        depends_lib port:py${python.version}-requests

        depends_test \
                    port:py${python.version}-pytest \
                    port:py${python.version}-mock
    }
```

when you do `depend_{lib, test}` you'll just overwrite whatever you've defined earlier. Sometimes if there is only one dependency that you don't need anymore, you can do `depends_lib-delete .....`, so the exact way of doing it depends a bit on the situation.

Reply:
the tests for the latest version fail miserably, did you run them locally? The first error I get is a missing dependency on `pymongo`, adding that one makes things even worse as the test-suite continues now, but pretty much everything fails. Can you please look into that?

Reply:
I did run the tests - they need to have several databases add which (mongo, redis, dynamodb, etc).  I can add them as dependencies for test.  Alternatively, I can just remove the test section.  I'm not sure what makes the most sense here.

Reply:
sorry - also forgot that gridfs and DynamoDB are going to be a problem with the test as they're no in macports

Reply:
that's fine - you don't need to have the test-phase in there if it requires a lot of additional ports... we should assume that upstream releases working versions and we do the correct thing with packaging it ;)

Reply:
removed in that recent force push